src/extensions/registries/status-bar-registry.ts

```diff
@@ -3,8 +3,12 @@
 import React from "react";
 import { BaseRegistry } from "./base-registry";
 
+export interface StatusBarComponents {
+  Item?: React.ComponentType<any>;
+}
+
 export interface StatusBarRegistration {
-  item?: React.ReactNode;
+  components: StatusBarComponents;
```

Comment:
It feels strange to have this name be plural. Even based on the type def, we're still talking about a _singular_ component. Whether that "component" is a fragment that's technically an array of components (but still, I believe, seen as a single component within the realms of React), or is a classical or functional single component, it's still just one component, right...?

Reply:
It's consistent across all the registration types. Each of the XXXRegistration interfaces now has a `components` field, which map to interfaces specifying one or more components:

```
export interface PageComponents {
  Page: React.ComponentType<any>;
}
```

```
export interface AppPreferenceComponents {
  Hint: React.ComponentType<any>;
  Input: React.ComponentType<any>;
}
```

better to use `components` for all rather than `component` for some and `components` for others. Or do you suggest changing all of them to `component`?

Reply:
Oh, I see. Consistency is key, for sure. What does `React.ComponentType<any>` represent, though. An array? If so, then I support the plural name. If not, why the "s"?

In any case, I'm probably fighting a losing battle since there is a lot of precedent if all other types are already using the plural form, and updating all of them would break a lot more stuff than just my extension.